nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EncryptAttributes.java

```diff
@@ -0,0 +1,508 @@
+                } else if (flowFileAttrs.contains(attr.key()) && attrsToEncryptPropVal.equals(CORE_ATTR)) {
+                    attrsToEncrypt.add(attr.key());
+                }
+            }
+        }
+
+        if (attrsToEncryptPropVal.equals(CUSTOM_ATTR)) {
+
+            //get list of all the attributes matching regex
+            if (regex != null && !regex.equals(".*")) {
+                attrsToEncrypt.clear();
+                Pattern pattern = Pattern.compile(regex);
+                for (String str : flowFileAttrs) {
+                    if (pattern.matcher(str).matches()) {
+                        attrsToEncrypt.add(str);
+                    }
+                }
+            }
+
+            //check if property-key is present in attrsToEncrypt and if expression-lang condition
+            //return true then encrypt/decrypt it.
+            if (!propMap.isEmpty()) {
+                HashSet<String> attrsToEncryptClone = new HashSet<>(attrsToEncrypt);
+                for(String attr: attrsToEncryptClone) {
+                    if (propMap.containsKey(attr)) {
+                        boolean matches = propMap.get(attr).evaluateAttributeExpressions(flowFile).asBoolean();
+                        if (!matches){
+                            attrsToEncrypt.remove(attr);
+                            getLogger().warn("{} expression-language expression evaluates to false",
+                                    new Object[]{propMap.get(attr).getValue()});
+                        }
+                    } else {
+                        attrsToEncrypt.remove(attr);
+                    }
+                }
+
+            }
+        }
+
+        attrsToEncrypt.remove(CoreAttributes.UUID.key());
+        if (EncryptProcessorUtils.isPGPAlgorithm(algorithm)) {
+            attrsToEncrypt.remove(CoreAttributes.FILENAME.key());
+            getLogger().info("Removing filename from {}cryption because of {} algorithm",
+                    new Object[]{(encrypt)?"en":"de", algorithm});
+        }
+        return attrsToEncrypt;
+    }
+
+    private Map<String, String> buildNewAttributes(FlowFile flowFile, PropertyValue attrList,
+                                                   PropertyValue attrSelectRegex, String algorithm,
+                                                   Encryptor encryptor, boolean encrypt) throws Exception {
+
+        Map<String, String> oldAttrs = flowFile.getAttributes();
+        Map<String, String> newAttrs = new HashMap<>();
+        Set<String> attrToEncrypt = getAttrToEncrypt(flowFile, attrList, attrSelectRegex, algorithm, encrypt);
+
+        for (String attr : attrToEncrypt) {
+            String attrVal = oldAttrs.get(attr);
+            String encryptedVal = (encrypt) ? performEncryption(attrVal, encryptor) : performDecryption(attrVal, encryptor);
+            newAttrs.put(attr, encryptedVal);
+            getLogger().debug("{}crypted {} from '{}' to '{}'",
```

Comment:
Having a message in the log that says "Attribute `x` was encrypted/decrypted" is fine. A message that says "Encrypted creditCardNum from `1234 5678 9012 3456` to `BxHF7a==`" is **absolutely not**. If someone is encrypting an attribute, it's probably because the value of the attribute is sensitive.